src/core/config.rs

```diff
@@ -14,20 +14,41 @@ pub fn get_core_hooks_path(repo: &Repo) -> eyre::Result<PathBuf> {
 }
 
 /// Get the configured name of the main branch.
+///
+/// The following config values are resolved, in order. The first valid value is returned.
+/// - branchless.core.mainBranch
+/// - (deprecated) branchless.mainBranch
+/// - init.defaultBranch
+/// - finally, default to "master"
 #[instrument]
 pub fn get_main_branch_name(repo: &Repo) -> eyre::Result<String> {
     let config = repo.get_readonly_config()?;
-    let main_branch_name: Option<String> = config.get("branchless.core.mainBranch")?;
-    let main_branch_name = match main_branch_name {
-        Some(main_branch_name) => main_branch_name,
-        None => {
-            // Deprecated; use `branchless.core.mainBranch` instead.
-            config
-                .get("branchless.mainBranch")?
-                .unwrap_or_else(|| "master".to_string())
-        }
+
+    if let Some(branch_name) = config.get("branchless.core.mainBranch")? {
+        return Ok(branch_name);
+    }
+
+    if let Some(branch_name) = config.get("branchless.mainBranch")? {
+        return Ok(branch_name);
+    }
+
+    if let Some(branch_name) = get_default_branch_name(repo)? {
+        return Ok(branch_name);
+    }
+
+    Ok("master".to_string())
```

Comment:
I had originally written this w/ nested matches, but I think they were 3 deep (b/c we had to unwrap 3 different config Options. I changed it to this which -- to me -- feels easier to grok at a glance. I might not be very rust-y, though. I can revert to something like that if you'd prefer